src/sys.rs

```diff
@@ -0,0 +1,349 @@
+        sys.wDay = tm.tm_mday as WORD;
+        sys.wDayOfWeek = tm.tm_wday as WORD;
+        sys.wMonth = (tm.tm_mon + 1) as WORD;
+        sys.wYear = (tm.tm_year + 1900) as WORD;
+        sys
+    }
+
+    fn system_time_to_tm(sys: &SYSTEMTIME, tm: &mut Tm) {
+        tm.tm_sec = sys.wSecond as i32;
+        tm.tm_min = sys.wMinute as i32;
+        tm.tm_hour = sys.wHour as i32;
+        tm.tm_mday = sys.wDay as i32;
+        tm.tm_wday = sys.wDayOfWeek as i32;
+        tm.tm_mon = (sys.wMonth - 1) as i32;
+        tm.tm_year = (sys.wYear - 1900) as i32;
+        tm.tm_yday = yday(tm.tm_year, tm.tm_mon + 1, tm.tm_mday);
+
+        fn yday(year: i32, month: i32, day: i32) -> i32 {
+            let leap = if month > 2 {
+                if year % 4 == 0 { 1 } else { 2 }
+            } else {
+                0
+            };
+            let july = if month > 7 { 1 } else { 0 };
+
+            (month - 1) * 30 + month / 2 + (day - 1) - leap + july
+        }
+    }
+
+
+ 
+    pub fn time_to_utc_tm(sec: i64, tm: &mut Tm) {
+        let mut out = unsafe { mem::zeroed() };
+        let ft = time_to_file_time(sec);
+        unsafe {
+            FileTimeToSystemTime(&ft, &mut out);
+        }
+        system_time_to_tm(&out, tm);
+        tm.tm_utcoff = 0;
+    }
+
+    pub fn time_to_local_tm(sec: i64, tm: &mut Tm) {
+        let mut out = unsafe { mem::zeroed() };
+        let tz = time_zone(None);
+        let ft = time_to_file_time(sec);
+        unsafe {
+            let mut utc = mem::zeroed();
+            FileTimeToSystemTime(&ft, &mut utc);
+            SystemTimeToTzSpecificLocalTime(tz, &mut utc, &mut out);
+            system_time_to_tm(&out, tm);
+            tm.tm_utcoff = -(*tz).Bias * 60;
+        }
+    }
+
+    pub fn utc_tm_to_time(tm: &Tm) -> i64 {
+        unsafe {
+            let mut ft = mem::zeroed();
+            let sys_time = tm_to_system_time(tm);
+            SystemTimeToFileTime(&sys_time, &mut ft);
+            file_time_to_time(&ft)
+        }
+    }
+
+    pub fn local_tm_to_time(tm: &Tm) -> i64 {
+        unsafe {
+            let mut ft = mem::zeroed();
+            let mut utc = mem::zeroed();
+            let mut sys_time = tm_to_system_time(tm);
+            TzSpecificLocalTimeToSystemTime(time_zone(None), &mut sys_time, &mut utc);
+            SystemTimeToFileTime(&utc, &mut ft);
+            file_time_to_time(&ft)
+        }
+    }
+
+    pub fn get_time() -> (i64, i32) {
+        unsafe {
+            let mut ft = mem::zeroed();
+            GetSystemTimeAsFileTime(&mut ft);
+            (file_time_to_time(&ft), file_time_to_nsec(&ft))
+        }
+    }
+
+    pub fn get_precise_ns() -> u64 {
+        let mut ticks = 0;
+        unsafe {
+            assert!(QueryPerformanceCounter(&mut ticks) == 1);
+        }
+        mul_div_i64(ticks as i64, 1000000000, frequency() as i64) as u64
+
+    }
+
+    pub fn time_zone(bias: Option<i32>) -> *const TIME_ZONE_INFORMATION {
```

Comment:
Hm ok! I'll add some comments, but I may remove this altogether and just update the tests.